Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -823,15 +819,32 @@ internal async Task PerformActionTypeBehavior(TreeNode treeNode, string treeNode
                 maxRetryCount--;
             }
 
+            if (actionTimeout != -1 && stopwatch.ElapsedMilliseconds + waitTime.TotalMilliseconds >= actionTimeout)
+            {
+                // If the timeout is hit and the ContinuationOnTimeout flag is set, commit a new ActionResponse. 
+                // with the status set to TimeoutOnAction and return.
+                if (treeAction.ContinuationOnTimeout)
+                {
+                    ActionResponse timeoutResponse = new ActionResponse
+                    {
+                        Status = "TimeoutOnAction"
+                    };
+
+                    await this.CommitActionResponse(treeActionKey, timeoutResponse).ConfigureAwait(false);
+                    return;
+                }
+            }
+
             // Retries are exhausted. Throw ActionTimeoutException with executeAction exception as innerException.
```

Comment:
>Retries are exhausted [](start = 15, length = 21)

Let's update this message as well since it should be hitting timeouts and not retries exhausted.